client/src/plugins/deployment-tool/__tests__/DeploymentToolSpec.js

```diff
@@ -84,6 +72,50 @@ describe('<DeploymentTool>', () => {
       onClose();
     });
 
+
+    it('should read and save config for deployed file', async () => {
+
+      // given
+      const config = sinon.stub({
+        getForFile() {
+          return {};
+        },
+        setForFile() {}
+      });
+      const details = {
+        endpointUrl: 'http://localhost:8088/engine-rest',
+        tenantId: '',
+        deploymentName: 'diagram',
+        authType: AuthTypes.basic,
+        username: 'demo',
+        password: 'demo'
+      };
+
+      const activeTab = createTab({ name: 'foo.bpmn' });
+      const {
+        wrapper,
+        instance
+      } = createDeploymentTool({ activeTab, config }, mount);
+
+      mounted = wrapper;
+
+      // when
+      instance.deploy();
+
+      await nextTick();
+      wrapper.update();
+
+      const { handleClose } = wrapper.state('modalState');
+
+      handleClose(details);
+
+      await nextTick();
+
+      // then
+      expect(config.getForFile).to.have.been.calledOnce;
+      expect(config.setForFile).to.have.been.calledOnce;
```

Comment:
Maybe let's also test for which file and with which details it got called?

Reply:
OK